### PR TITLE
refactor: change explorer provider state into saved chart version

### PR DESCRIPTION
**Pull request by ZeRego**
### Pre requirements:
- [x] I have followed (at least) the [PR section of the contributing guide](https://github.com/lightdash/lightdash/blob/main/.github/CONTRIBUTING.md#sending-a-pull-request).

### Reference:
Closes: 

### Description:

- make the explorer provider state as close as possible as the saved chart version

### Scope of the changes (select all that apply):
- [x] Frontend  
- [ ] Backend  
- [ ] Common  
- [ ] E2E tests  
- [ ] Docs  
- [ ] CI/CD

### CODE REVIEW

**Comment by rephus** [comment]
A few e2e tests are failing, including this one (I've never seen it fail) 
image

Could you check if it is something to do with your branch ? if it is not, or e2e works on your machine, I'll take a look if it keeps happening on main

**Comment by ZeRego** [comment]
@rephus reran e2e tests here https://github.com/lightdash/lightdash/runs/6154417014?check_suite_focus=true and they all passed

File: packages/frontend/src/providers/ExplorerProvider.tsx [hunk 1]
```diff
@@ -147,23 +138,61 @@ interface ExplorerContext {
...
+            additionalMetrics: [],
+        },
+        pivotConfig: {
+            columns: [],
+        },
+        tableConfig: {
+            columnOrder: [],
+        },
+        chartConfig: {
+            type: ChartType.CARTESIAN,
```
[the remark this answers is not in the document]
Comment (ZeRego) [reply packages/frontend/src/providers/ExplorerProvider.tsx:164]: This is the default state. There are actions to change the chart type and config. 
Supporting multiple configurations is out of scope for this refactor.

File: packages/common/src/types/savedCharts.ts [hunk 2]
```diff
@@ -128,6 +128,15 @@ export const isCompleteEchartsConfig = (
 ): value is CompleteEChartsConfig =>
     !!value && !!value.series && value.series.length > 0;
 
+export const isCartesianChartConfig = (
```
[the remark this answers is not in the document]
Comment (ZeRego) [reply packages/common/src/types/savedCharts.ts:131]: I agree but currently our code is handling the type and config separately so we need these safeguards. Further work is needed for our code to handle the type+config as one.

**Pull request merged**